### PR TITLE
Update Application documentation.

### PR DESCRIPTION
**Pull request by jmeas**
This is a rough draft of some much-needed updates to the app docs that should hopefully clarify the purpose of the Class.

Do note that there are some things we might want to fix up. Grammar and such. I wrote this pretty quickly just to get the idea on paper.

### CODE REVIEW

File: docs/marionette.application.md [hunk 1]
```diff
@@ -2,21 +2,34 @@
...
+
+- A namespace to keep things off of the `window`.
+  If you are not using a module loader like ES6 modules, CommonJS, or AMD, then
+  you can use the Application to store your Javascript objects. And if you are
+  using one of those module systems, then you can still attach things to the
+  application to aid in debugging.
+
+- Integration with the Marionette Inspector. The Marionette Inspector is a fantastic tool
+  that makes it easy to understand and debug your application. Using the Application Class
+  will automatically hook up your application to that extension.
```
Comment (jmeas) [inline docs/marionette.application.md]: Is this true? Or does it need to be on `window.app`? //cc @jasonLaster

File: docs/marionette.application.md [hunk 2]
```diff
@@ -2,21 +2,34 @@
 
 # Marionette.Application
 
-The `Backbone.Marionette.Application` object is the hub of your composite
-application. It organizes, initializes and coordinates the various pieces of your
-app. It also provides a starting point for you to call into from your HTML
-script block, or directly from your JavaScript files if you prefer to go that
-route.
+The **Application** is a container for the rest of your code. It is recommended
```
Comment (megawac) [inline docs/marionette.application.md]: > The **Application** is a container

& bootstrapper?

Comment (jmeas) [inline docs/marionette.application.md]: I like the sentiment, but I wonder if I can communicate it in a different way. 'Bootstrapper' seems too colloquial. Nawm sayin?

File: docs/marionette.application.md (hunk 1 above)
Comment (jasonLaster) [inline docs/marionette.application.md]: nope - just needs to exist. And honestly, I'd like to remove the `app` dependency entirely. So, I'm not sure I really need this paragraph

**Comment by jasonLaster** [comment]
I like where this is headed @jmeas - I'll provide more thoughts later

File: docs/marionette.application.md [hunk 3]
```diff
@@ -2,21 +2,34 @@
...
+that every Marionette app have at least one instance of Application.
```
Comment (thejameskyle) [inline docs/marionette.application.md]: Isn't it only possible to have one application today?

Comment (samccone) [inline docs/marionette.application.md]: you can have multiple, just not multiple routers :)

Comment (thejameskyle) [inline docs/marionette.application.md]: ...but i have multiple routers...

Comment (jmeas) [inline docs/marionette.application.md]: If you look through the source, it's easy to confirm that you can have any number of Applications or Routers. The only thing to be wary of is a channel conflict, but changing the channel name is documented, and using the channel is deprecated.

Comment (thejameskyle) [inline docs/marionette.application.md]: Aha! That's what I was trying to remember

**Comment by samccone** [comment]
:ship: 'd

**Pull request closed without merging**